chall07/pcuadrad.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"strconv"
+	"log"
+	"os"
+	"time"
+)
+
+func RGBToHex(r int, g int, b int) (string) {
+	return fmt.Sprintf("%x%x%x",r ,g ,b)
+}
+
+func printRequestTime(url string, answer string, start time.Time) {
+	now := time.Now()
+	elapsed := now.Sub(start)
+	if answer == "" {
+		fmt.Println(elapsed,"- GET", url)
+	} else {
+		fmt.Println(elapsed,"- GET", url, "\n\tanswer:", answer)
+	}
+}
+
+func DoRequest(url string, l *log.Logger, start time.Time) (string) {
+	printRequestTime(url, "", start)
+	response, err := http.Get(url)
+	if err != nil {
+		l.Fatalln(err)
+	}
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		l.Fatalln(err)
+	}
+	printRequestTime(url, string(body), start)
+	return string(body)
+}
+
+func parseBody(body string, l *log.Logger) (string, int, int, int){
+	var arr [4]int
+	var err error
+	parse := strings.SplitN(body, " ", 2)
+	parse = strings.Split(parse[0], ",")
+	for i := 0; i < len(parse); i++ {
+		getNumbers := strings.Split(parse[i], "=")
+		arr[i], err = strconv.Atoi(getNumbers[1])
+		if err != nil {
+			l.Fatalln(err)
+		}
+	}
+	return strconv.Itoa(arr[0]), arr[1], arr[2], arr[3]
```

Comment:
This is a nice parsing, however there is most simple way of doing this. You could have used [fmt.Sscanf](https://golang.org/pkg/fmt/#Sscanf). As you know exactly how the input should be:
```suggestion
	var id, red, green, blue int
	if _, err := fmt.Sscanf(body, "id=%d,r=%d,g=%d,b=%d - Send your response here: chall07.42madrid.com/?id=<id>&resp=<hex>", &id, &red, &green, &blue); err != nil {
		l.Fatalln(err)
	}
	return  strconv.Itoa(id), red, green, blue
```
Although I don't really understand the need of converting `id` into a string, I let it be this way.

Reply:
I didn't know the existence of that function, so it is so much simpler.
The conversion of the id into string, was more than anything because to concatenate it later in the url string I had to parse string(id) and I wanted it to be as clean as possible.